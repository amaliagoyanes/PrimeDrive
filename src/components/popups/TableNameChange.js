import React, {Component} from 'react';
import {
    Dimensions,
    TouchableOpacity,
    Text,
    Button,
    View,
    TextInput,
} from 'react-native';
import {FormattedMessage} from 'react-intl';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {Row, Col} from 'react-native-easy-grid';
import Modal from 'react-native-root-modal';
import styles from '../../style/styles';
import {scale, scaleText} from '../../scaling';


export default class TableNameChange extends Component {
    constructor(props){
        super(props);
        this.state = {
            open: false,
            table: {},
            input: props.value || ''
        };

        this.close = this.close.bind(this);
        this.open = this.open.bind(this);
        this.okOnPress = this.okOnPress.bind(this);
    }

    open(table){
        this.setState({open: true, table: table})
    }

    close(){
        this.setState({open: false, table: {}, input: ''})
    }

    okOnPress () {
        this.props.onOk(this.state.table, this.state.input);
        this.close();
    }

    render() {
        return (
            <Modal
                onRequestClose={this.close}
                visible={this.state.open}
                style={styles.modal}
            >
                <TouchableOpacity style={styles.overlay} onPress={this.close}>
                    <TouchableOpacity activeOpacity={1} onPress={(event) => event.stopPropagation()} style={{
                        position: 'absolute',
                        top: Dimensions.get('screen').height * 0.175,
                        left: Dimensions.get('screen').width * 0.375,
                        width: Dimensions.get('screen').width * 0.25,
                        height: Dimensions.get('screen').height * 0.2,
                        backgroundColor: '#e3e5e6'
                    }}>
                        <Row size={0.5} style={{alignItems: 'flex-start', justifyContent: 'flex-end'}}>
                            <TouchableOpacity onPress={this.close}
                                              style={[styles.timesIcon, {marginTop: 0, marginLeft: scale(18), backgroundColor: 'transparent'}]}>
                                <Icon size={scaleText(20)} name="close" color="#313138"/>
                            </TouchableOpacity>
                        </Row>
                        <Row size={1.5} style={{paddingLeft: scale(10), paddingRight: scale(10)}}>
                            <Col>
                                <Row>
                                    <Text style={[styles.text, {color: '#313138', textAlign: 'left'}]}>
                                        {this.props.intl.messages[this.props.message ? this.props.message : "app.changeTableName"]}</Text>
                                </Row>
                                <Row>
                                    <TextInput onChange={(event) => this.setState({input: event.nativeEvent.text})}
                                               style={[styles.text, styles.tableNameChangeInput]} value={this.state.input}
                                               autoFocus={true}/>
                                </Row>
                            </Col>

                        </Row>
                        <Row style={{alignItems: 'flex-end', justifyContent: 'flex-end'}}>
                            <TouchableOpacity onPress={this.close} style={[styles.tabButton, styles.confirmBtn,
                                {backgroundColor: 'transparent',}]}>
                                <Text style={[styles.text, {color: '#313138', fontSize: scaleText(13)}]}>
                                    {this.props.intl.messages["app.cancelUpper"]}</Text>
                            </TouchableOpacity>
                            <TouchableOpacity onPress={this.okOnPress}
                                              style={[styles.tabButton, styles.confirmBtn]}>
                                <Text style={[styles.text, {fontSize: scaleText(13)}]}>
                                    {this.props.intl.messages["app.okUpper"]}
                                </Text>
                            </TouchableOpacity>
                        </Row>
                    </TouchableOpacity>
                </TouchableOpacity>
            </Modal>
        )
    }
}
