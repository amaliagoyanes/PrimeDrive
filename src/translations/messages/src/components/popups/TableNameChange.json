[
  {
    "id": "app.changeTableName",
    "defaultMessage": "Change table name to:"
  }
]