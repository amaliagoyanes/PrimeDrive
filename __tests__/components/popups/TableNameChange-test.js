'use strict';

import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import TableNameChange from '../../../src/components/popups/TableNameChange';


it('renders correctly', () => {
    const tree = shallow(
        <TableNameChange />);
    tree.setState({open: true});
    expect(tree).toMatchSnapshot();
    expect(tree.find('TextInput').length).toEqual(1)
});