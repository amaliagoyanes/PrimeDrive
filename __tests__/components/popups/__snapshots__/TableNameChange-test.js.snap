// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  onRequestClose={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.2)",
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  visible={true}
>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,0.1)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={1}
      focusedOpacity={0.7}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#e3e5e6",
          "height": 266.8,
          "left": 281.25,
          "position": "absolute",
          "top": 500.25,
          "width": 187.5,
        }
      }
    >
      <RowNB
        size={0.5}
        style={
          Object {
            "alignItems": "flex-start",
            "justifyContent": "flex-end",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          focusedOpacity={0.7}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#3f3f48",
                "height": 24,
                "justifyContent": "center",
                "width": 24,
              },
              Object {
                "backgroundColor": "transparent",
                "marginLeft": 18,
                "marginTop": 0,
              },
            ]
          }
        >
          <Icon
            allowFontScaling={false}
            color="#313138"
            name="close"
            size={20}
          />
        </TouchableOpacity>
      </RowNB>
      <RowNB
        style={
          Object {
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <ColumnNB>
          <RowNB>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontFamily": "Gotham-Book",
                    "fontSize": 14,
                  },
                  Object {
                    "color": "#313138",
                    "textAlign": "left",
                  },
                ]
              }
            >
              Change table name to:
            </Text>
          </RowNB>
          <RowNB>
            <TextInput
              autoFocus={true}
              onChange={[Function]}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontFamily": "Gotham-Book",
                    "fontSize": 14,
                  },
                  Object {
                    "color": "#313138",
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
            />
          </RowNB>
        </ColumnNB>
      </RowNB>
      <RowNB
        style={
          Object {
            "alignItems": "flex-end",
            "justifyContent": "flex-end",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          focusedOpacity={0.7}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#404048",
                "justifyContent": "center",
                "width": "34%",
              },
              Object {
                "backgroundColor": "#2dab61",
                "height": 45,
                "width": "50%",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Gotham-Book",
                  "fontSize": 14,
                },
                Object {
                  "color": "#313138",
                  "fontSize": 13,
                },
              ]
            }
          >
            CANCEL
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
          focusedOpacity={0.7}
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#404048",
                "justifyContent": "center",
                "width": "34%",
              },
              Object {
                "backgroundColor": "#2dab61",
                "height": 45,
                "width": "50%",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Gotham-Book",
                  "fontSize": 14,
                },
                Object {
                  "fontSize": 13,
                },
              ]
            }
          >
            OK
          </Text>
        </TouchableOpacity>
      </RowNB>
    </TouchableOpacity>
  </TouchableOpacity>
</Modal>
`;
